ui/app/src/utils/mimeTypes.ts

```diff
@@ -0,0 +1,212 @@
+  },
+  'text/texmacs': {
+    id: 'mimeTypes.text/texmacs',
+    defaultMessage: 'Typescript (ts)'
+  },
+  'text/css': {
+    id: 'mimeTypes.text/css',
+    defaultMessage: 'Cascading Style Sheets (CSS)'
+  },
+  'text/x-sass': {
+    id: 'mimeTypes.text/x-sass',
+    defaultMessage: 'Sass style sheet'
+  },
+  'text/x-scss': {
+    id: 'mimeTypes.text/x-scss',
+    defaultMessage: 'Scss stylesheet'
+  },
+  'text/x-groovy': {
+    id: 'mimeTypes.text/x-groovy',
+    defaultMessage: 'Groovy'
+  },
+  'text/x-freemarker': {
+    id: 'mimeTypes.text/x-freemarker',
+    defaultMessage: 'Freemarker'
+  },
+  'audio/aac': {
+    id: 'mimeTypes.audio/aac',
+    defaultMessage: 'AAC audio'
+  },
+  'audio/midi audio/x-midi': {
+    id: 'mimeTypes.audio/midi audio/x-midi',
+    defaultMessage: 'Musical Instrument Digital Interface (MIDI)'
+  },
+  'audio/mpeg': {
+    id: 'mimeTypes.audio/mpeg',
+    defaultMessage: 'MP3 audio'
+  },
+  'audio/wav': {
+    id: 'mimeTypes.audio/wav',
+    defaultMessage: 'Waveform Audio Format'
+  },
+  'video/x-msvideo': {
+    id: 'mimeTypes.video/x-msvideo',
+    defaultMessage: 'AVI: Audio Video Interleave'
+  },
+  'video/mp4': {
+    id: 'mimeTypes.video/mp4',
+    defaultMessage: 'MP4 video'
+  },
+  'video/mpeg': {
+    id: 'mimeTypes.video/mpeg',
+    defaultMessage: 'MPEG Video'
+  },
+  'application/xml': {
+    id: 'mimeTypes.application/xml',
+    defaultMessage: 'XML'
+  },
+  'text/html': {
+    id: 'mimeTypes.text/html',
+    defaultMessage: 'HyperText Markup Language (HTML)'
+  },
+  'text/plain': {
+    id: 'mimeTypes.text/plain',
+    defaultMessage: 'Text'
+  },
+  'application/vnd.ms-fontobject': {
+    id: 'mimeTypes.application/vnd.ms-fontobject',
+    defaultMessage: 'MS Embedded OpenType fonts'
+  },
+  'font/otf': {
+    id: 'mimeTypes.font/otf',
+    defaultMessage: 'OpenType font'
+  },
+  'font/ttf': {
+    id: 'mimeTypes.font/ttf',
+    defaultMessage: 'TrueType Font'
+  },
+  'font/woff': {
+    id: 'mimeTypes.font/woff',
+    defaultMessage: 'Web Open Font Format (WOFF)'
+  },
+  'font/woff2': {
+    id: 'mimeTypes.font/woff2',
+    defaultMessage: 'Web Open Font Format (WOFF)'
+  },
+  'application/gzip': {
+    id: 'mimeTypes.application/gzip',
+    defaultMessage: 'GZip Compressed Archive'
+  },
+  'application/x-7z-compressed': {
+    id: 'mimeTypes.application/x-7z-compressed',
+    defaultMessage: '7-zip archive'
+  },
+  'application/x-bzip': {
+    id: 'mimeTypes.application/x-bzip',
+    defaultMessage: 'BZip archive'
+  },
+  'application/x-bzip2': {
+    id: 'mimeTypes.application/x-bzip2',
+    defaultMessage: 'BZip2 archive'
+  },
+  'application/zip': {
+    id: 'mimeTypes.application/zip',
+    defaultMessage: 'ZIP archive'
+  },
+  'application/vnd.rar': {
+    id: 'mimeTypes.application/vnd.rar',
+    defaultMessage: 'RAR archive'
+  },
+  'application/x-tar': {
+    id: 'mimeTypes.application/x-tar',
+    defaultMessage: 'Tape Archive (TAR)'
+  }
+});
```

Comment:
@russdanner @sumerjabri please review. What's on defaultMessage is what users would see. I think for example, PDF should simply be PDF.